node/src/blockchain/test_utils.rs

```diff
@@ -101,65 +98,47 @@ impl BlockchainInterface for BlockchainInterfaceMock {
         self.retrieve_transactions_results.borrow_mut().remove(0)
     }
 
-    fn mobilize_payable_payments_agent(&self) -> Box<dyn PayablePaymentsAgent> {
-        self.mobilize_payable_payments_agent_results
-            .borrow_mut()
-            .remove(0)
+    fn build_blockchain_agent(
+        &self,
+        consuming_wallet: &Wallet,
+        persistent_config: &dyn PersistentConfiguration,
+    ) -> Result<Box<dyn BlockchainAgent>, String> {
+        self.build_blockchain_agent_params.lock().unwrap().push((
+            consuming_wallet.clone(),
+            persistent_config.arbitrary_id_stamp(),
+        ));
+        self.build_blockchain_agent_results.borrow_mut().remove(0)
     }
 
     fn send_batch_of_payables(
         &self,
-        consuming_wallet: &Wallet,
-        payable_payments_agent: &dyn PayablePaymentsAgent,
+        agent: Box<dyn BlockchainAgent>,
         new_fingerprints_recipient: &Recipient<PendingPayableFingerprintSeeds>,
         accounts: &[PayableAccount],
     ) -> Result<Vec<ProcessedPayableFallible>, PayableTransactionError> {
         self.send_batch_of_payables_params.lock().unwrap().push((
-            consuming_wallet.clone(),
-            payable_payments_agent.arbitrary_id_stamp(),
+            agent.arbitrary_id_stamp(),
             new_fingerprints_recipient.clone(),
             accounts.to_vec(),
         ));
         self.send_batch_of_payables_results.borrow_mut().remove(0)
     }
 
-    fn get_transaction_fee_balance(&self, address: &Wallet) -> ResultForBalance {
-        self.get_transaction_fee_balance_params
-            .lock()
-            .unwrap()
-            .push(address.clone());
-        self.get_transaction_fee_balance_results
-            .borrow_mut()
-            .remove(0)
-    }
-
-    fn get_token_balance(&self, address: &Wallet) -> ResultForBalance {
-        self.get_token_balance_params
-            .lock()
-            .unwrap()
-            .push(address.clone());
-        self.get_token_balance_results.borrow_mut().remove(0)
-    }
-
-    fn get_transaction_count(&self, wallet: &Wallet) -> ResultForNonce {
-        self.get_transaction_count_parameters
-            .lock()
-            .unwrap()
-            .push(wallet.clone());
-        self.get_transaction_count_results.borrow_mut().remove(0)
-    }
-
     fn get_transaction_receipt(&self, hash: H256) -> ResultForReceipt {
         self.get_transaction_receipt_params
             .lock()
             .unwrap()
             .push(hash);
         self.get_transaction_receipt_results.borrow_mut().remove(0)
     }
+
+    fn helpers(&self) -> &dyn RPCHelpers {
+        intentionally_blank!()
+    }
 }
 
 impl BlockchainInterfaceMock {
-    //seems like never-used
+    // TODO seems like never-used
```

Comment:
This is pretty simple. If it isn't used, you can probably delete it with confidence that someone who needs it in the future can write it again.